src/components/Form.js

```diff
@@ -138,10 +136,13 @@ export default class Form extends Component {
       autocomplete,
       enctype,
       acceptcharset,
-      noHtml5Validate
+      noHtml5Validate,
+      idSchema: originalIdSchema
     } = this.props;
 
-    const {schema, uiSchema, formData, errorSchema, idSchema} = this.state;
+    const {schema, uiSchema, formData, errorSchema} = this.state;
+    const idSchema = originalIdSchema || toIdSchema(schema, uiSchema["ui:rootFieldId"], schema.definitions);
```

Comment:
I've made it so that `idSchema` is no longer exclusively determined by state. If an `idSchema` prop is passed in, that will be used. Otherwise, `idSchema` will be determined using the state variables just as it was previously.